Code/GraphMol/RWMol.cpp

```diff
@@ -127,19 +127,24 @@ unsigned int RWMol::addAtom(bool updateLabel) {
   return rdcast<unsigned int>(which);
 }
 
-void RWMol::replaceAtom(unsigned int idx, Atom *atom_pin, bool updateLabel) {
+void RWMol::replaceAtom(unsigned int idx, Atom *atom_pin, bool updateLabel,
+                        bool preserveProps) {
   RDUNUSED_PARAM(updateLabel);
   PRECONDITION(atom_pin, "bad atom passed to replaceAtom");
   URANGE_CHECK(idx, getNumAtoms() - 1);
   Atom *atom_p = atom_pin->copy();
   atom_p->setOwningMol(this);
   atom_p->setIdx(idx);
   MolGraph::vertex_descriptor vd = boost::vertex(idx, d_graph);
+  if (preserveProps) {
+    const bool replaceExistingData = false;
```

Comment:
great. Only outstanding thing is to add support for this to the python wrappers (see above)